src/address.rs

```diff
@@ -135,44 +134,59 @@ impl<A, Rc: RefCounter> Address<A, Rc> {
     /// Convert this address into a generic address which can be weak or strong.
     pub fn as_either(&self) -> Address<A, Either> {
         Address {
-            sink: self.sink.clone(),
+            sender: self.sender.clone(),
             ref_counter: self.ref_counter.clone().into_either(),
         }
     }
 
-    /// Send a message to the actor.
-    ///
-    /// The actor must implement [`Handler<Message>`] for this to work.
-    ///
-    /// This function returns a [`Future`](SendFuture) that resolves to the [`Return`](crate::Handler::Return) value of the handler.
-    /// The [`SendFuture`] will resolve to [`Err(Disconnected)`] in case the actor is stopped and not accepting messages.
-    #[allow(clippy::type_complexity)]
-    pub fn send<M>(
-        &self,
-        message: M,
-    ) -> SendFuture<
-        <A as Handler<M>>::Return,
-        NameableSending<A, <A as Handler<M>>::Return>,
-        ResolveToHandlerReturn,
-    >
-    where
-        M: Send + 'static,
-        A: Handler<M>,
+    /// TODO(bounded)
+    pub fn do_send<M>(&self, message: M) -> Result<(), Disconnected>
+        where A: Handler<M>,
+              M: Send + 'static
     {
-        if self.is_connected() {
-            let (envelope, rx) = ReturningEnvelope::<A, M, <A as Handler<M>>::Return>::new(message);
-            let tx = self
-                .sink
-                .sender()
-                .clone()
-                .into_send_async(AddressMessage::Message(Box::new(envelope)));
+        let (envelope) = NonReturningEnvelope::<A, M>::new(message);
+        self.sender.send(Box::new(envelope), 1).map_err(|_| Disconnected)
+    }
 
-            SendFuture::sending_named(tx, rx)
-        } else {
-            SendFuture::disconnected()
-        }
+    /// TODO(bounded)
+    pub async fn send<M>(&self, message: M) -> Result<<A as Handler<M>>::Return, Disconnected>
+        where A: Handler<M>,
+              M: Send + 'static,
+    {
+        let (envelope, rx) = ReturningEnvelope::<A, M, <A as Handler<M>>::Return>::new(message);
+        self.sender.send(Box::new(envelope), 0).map_err(|_| Disconnected)?;
+        rx.await.map_err(|_| Disconnected)
     }
```

Comment:
Is the existence of `do_send` and `send` an indicator that you'd like to move away again from a unified `send` API? That would be a bummer, especially if `do_send` retains its current signature.

I would strongly recommend to at least limit `do_send` to `Handler`s which return `()`, otherwise, this becomes a massive footgun where potential errors within `Result` return types are dropped without the user noticing. Dropping the concept of `NonReturningEnvelope`s entirely would be even better IMO.

As demonstrated by https://github.com/Restioson/xtra/pull/74, the concepts of "only block (the task) until message is queued" and "access the return value of the handler" are actually orthogonal.

Reply:
> Is the existence of do_send and send an indicator that you'd like to move away again from a unified send API? That would be a bummer, especially if do_send retains its current signature.

This was a temporary change since I haven't yet implemented bounding yet at all :) I do want to go back to the old API

Reply:
Cool, I figured that it was something like this, just wanted to clarify :)

Reply:
No problem :) Expect many other changes in here like that for now, haha... They should thin out as I implement more functionality in the `inbox` module.

Reply:
I've restored the original interface now :)